File: src/main/kotlin/dev/hawu/plugins/queuedsudo/commands/BaseCommand.kt
package dev.hawu.plugins.queuedsudo.commands

import dev.hawu.plugins.api.dsl.commands.CommandSpec
import dev.hawu.plugins.queuedsudo.I18n.tl
import org.bukkit.plugin.java.JavaPlugin

class BaseCommand(private val plugin: JavaPlugin) : CommandSpec({
    
    name.set("queuedsudo")
    permission.set("queued-sudo.main")
    permissionMessage.set("no-permissions".tl("perm" to permission.get()))
    
    "create" bindTo CreateCommand(plugin)
    "run" bindTo RunCommand(plugin)
    "reload" bindTo ReloadCommand()
    "search" bindTo SearchCommand(plugin)
    "list" bindTo ListCommand(plugin)
    "rename" bindTo RenameCommand(plugin)
    "delete" bindTo DeleteCommand(plugin)
    "addworld" bindTo AddWorldCommand(plugin)
    "removeworld" bindTo RemoveWorldCommand(plugin)
    
    on command { sender, _ ->
        sender.sendMessage("help-index".tl("version" to plugin.description.version))
        true
    }
    
    on tab { _, args -> if(args.size == 1) listOf(
        "create",
        "run",
        "reload",
        "search",
        "list",
        "rename",
        "delete",
        "addworld",
        "removeworld",
    ).filter { it.startsWith(args[0]) } else null }
    
    register to plugin
    
})

File: src/main/scala/dev/hawu/plugins/queuedsudo/BaseCommand.scala
package dev.hawu.plugins.queuedsudo

import dev.hawu.plugins.api.commands.{AbstractCommandClass, CommandArgument, CommandSource}
import dev.hawu.plugins.queuedsudo.commands.{GroupsCommand, ReloadCommand}
import dev.hawu.plugins.queuedsudo.LocalI18n.*
import org.bukkit.plugin.java.JavaPlugin

import java.util
import scala.jdk.CollectionConverters.*

/**
 * Command handler for the main command of the plugin.
 *
 * @param pl the plugin responsible
 */
class BaseCommand(private val pl: JavaPlugin) extends AbstractCommandClass("queuedsudo") :

   allowAny()
   setPermission("queued-sudo.main")
   register(pl)

   bind(GroupsCommand())
   bind(ReloadCommand())

   override def run(sender: CommandSource, args: CommandArgument): Unit = sender.tl("help.base")

   override def tab(sender: CommandSource, args: CommandArgument): util.List[String] = List(
      "groups",
      "reload",
   ).asJava
